modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -233,6 +234,8 @@ public class PartitionReplicaListener implements ReplicaListener {
 
     private final Supplier<Map<Integer, IndexLocker>> indexesLockers;
 
+    private final ConcurrentMap<UUID, CompletableFuture<Void>> txDurableFinishFutures = new ConcurrentHashMap<>();
```

Comment:
Please add an explanation why we need given futures. And by the way (I didn't check it yet) why we really need them? 
Finish operation is idempotent, we will retry it rather rare.